test_fms/diag_manager/test_diag_update_buffer.F90

```diff
@@ -0,0 +1,519 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @brief  This programs tests the update of field data buffers with
+!! the  "math" functions in module fms_diag_fieldbuff_update_mod. It mimics
+!! the daig_manager::send_4d operation of calling those functions.
+program test_diag_update_buffer
+   use platform_mod
+   use mpp_mod, only: mpp_init, mpp_set_stack_size, mpp_init_test_requests_allocated
+   use fms_mod, ONLY: fms_init, fms_end, error_mesg, FATAL,NOTE
+   use diag_data_mod, ONLY: fms_diag_ibounds_type, VERY_LARGE_AXIS_LENGTH
+   USE fms_diag_outfield_mod, ONLY: fms_diag_outfield_type, fms_diag_outfield_index_type
+   USE fms_diag_fieldbuff_update_mod, ONLY: fieldbuff_update, fieldbuff_copy_misvals, &
+   & fieldbuff_copy_fieldvals
+   USE fms_diag_time_reduction_mod, ONLY: time_reduction_type, time_average, time_rms
+
+   implicit  none
+
+   !! Class diag_buffer_type is here only for temporary use for modern diag_manager
+   !! development until the real buffer class is sufficiently ready and merged.
+   TYPE diag_test_buffer_type
+      CLASS(*), ALLOCATABLE, DIMENSION(:,:,:,:,:) :: buffer
+      CLASS(*), ALLOCATABLE, DIMENSION(:,:,:,:,:) :: counter
+      CLASS(*), ALLOCATABLE, DIMENSION(:)  :: count_0d
+      INTEGER, ALLOCATABLE, dimension(:) :: num_elements
+   END TYPE diag_test_buffer_type
+
+   integer,parameter :: SZ=10    !<Field data this size in all spatiall dims.
+   integer,parameter :: SL=2     !!Field data this size in 4th dim
+   integer,parameter :: NDI=1    !!Number of diurnal elemes
+   CLASS(*), ALLOCATABLE :: r4_datapoint, i8_datapoint !> to be allocated of rype data (e.g. r4. i8)
```

Comment:
_!>_ pointing in the wrong direction!

Reply:
You can actually do either way with doxygen, the `!<` actually tends to work better with variable declarations.

The doxygen stuff from the style guide is expanded on a bit here btw: http://noaa-gfdl.github.io/FMS/md_docs_doxygenGuide.html

Reply:
done